src/genetic_search_iterator.jl

```diff
@@ -0,0 +1,108 @@
+using Random
+
+Base.@kwdef struct GeneticSearchIterator{FitnessFunction,CrossOverFunction,MutationFunction,SelectParentsFunction,EvaluationFunction} <: ExpressionIterator
+    grammar::ContextSensitiveGrammar
+    examples::Vector{<:Example}
+
+    fitness::FitnessFunction
+    cross_over::CrossOverFunction
+    mutation!::MutationFunction
+    select_parents::SelectParentsFunction # selects two parents to crossover
+    evaluation_function::EvaluationFunction
+    
+    start_symbol::Symbol
+    population_size::Int64
+    mutation_probability::Float64
+    maximum_initial_population_depth::Int64
+
+    max_depth::Int64  = 10    # not used
+end
+
+struct GeneticIteratorState
+    population::Array{RuleNode}
+end
+
+Base.IteratorSize(::GeneticSearchIterator) = Base.SizeUnknown()
+Base.eltype(::GeneticSearchIterator) = RuleNode
```

Comment:
is this `RuleNode` or `Array{RuleNode}`?

Reply:
Right now is just `RuleNode`.